opensearch-project/OpenSearch-Dashboards
# [Workspace] Dashboard admin(groups/users) implementation

### Description
This PR set the backend roles in groups or users, whoever has the backend roles or exactly match the user or groups ids defined in this yml.config/dynamic config will be regard as dashboard admin.
Dashboard admin will have the access to all the workspaces and objects inside OpenSearch Dashboards.


### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6389



## Screenshot

No UI change in this PR.

## Testing the changes
> [!TIP]
> There are two ways to open the OSD configuration, so two test flows are provided respectively.
### Yml config test
1. Clone the latest code and clone security-dashboards-plugin to the plugins directory
2. Build the security-dashboards-plugin and bootstrap
3. Download the patch file [test.patch](https://github.com/opensearch-project/OpenSearch-Dashboards/files/14975232/test.patch) and yarn `git apply ./test.patch`
4. Modify `config/opensearch_dashboards.yml`
```yml
workspace.enabled: true 
opensearch.hosts: ["https://localhost:9200"]
opensearch.username: "username"
opensearch.password: "password" 
opensearch.ssl.verificationMode: none
savedObjects.permission.enabled: true
```
5. Two tests ways
   * **Yml config**
      1. Modify `config/opensearch_dashboards.yml`
      ```yml
      opensearchDashboards.dashboardAdmin.groups: ["dashboard_admin"]
      opensearchDashboards.dashboardAdmin.users: ["dashboard_admin"]
      ```
   * **Dynamic application config**
      1. Modify `config/opensearch_dashboards.yml`
      ```yml
      application_config.enabled: true
      ```
      2. Update the configuration of an entity by CURL command
      ```
      curl --insecure {osd endpoint}/api/appconfig/opensearchDashboards.dashboardAdmin.users -X POST -H 'Accept: application/json' -H 'Content-Type: application/json' -H 'osd-xsrf: osd-fetch' -H 'Sec-Fetch-Dest: empty' --data-raw '{"newValue":"[\"dashboard_admin\"]"}' -u 'username:password'
     ```
6. Run `yarn start --no-base-path` and visit http://localhost:5601
7. You can check whether it is dashboards admin in the console.
<img width="1617" alt="image" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/71875999/c64d84ea-2b1f-4d2a-a440-98ce7fa32e2a">


### Check List
